From 5162d06dd8ead04461f5c21ae4830ec41e528a59 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Thu, 15 Jan 2015 21:05:12 +0800
Subject: [PATCH 1313/1594] MLK-10102-6 usb: chipidea: otg: export
 ci_handle_id_switch API

commit e130afe623307b69b3737cb5a41905400082ca36 from
git://git.freescale.com/imx/linux-2.6-imx.git

Export ci_handle_id_switch interface for controller handle id
changes during system sleep with power lost.

Acked-by: Peter Chen <peter.chen@freescale.com>
Signed-off-by: Li Jun <b47624@freescale.com>
---
 drivers/usb/chipidea/otg.c |    2 +-
 drivers/usb/chipidea/otg.h |    1 +
 2 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/usb/chipidea/otg.c b/drivers/usb/chipidea/otg.c
index 7db6ec49..a016a89 100644
--- a/drivers/usb/chipidea/otg.c
+++ b/drivers/usb/chipidea/otg.c
@@ -82,7 +82,7 @@ void ci_handle_vbus_change(struct ci_hdrc *ci)
 }
 
 #define CI_VBUS_STABLE_TIMEOUT_MS 5000
-static void ci_handle_id_switch(struct ci_hdrc *ci)
+void ci_handle_id_switch(struct ci_hdrc *ci)
 {
 	enum ci_role role = ci_otg_role(ci);
 
diff --git a/drivers/usb/chipidea/otg.h b/drivers/usb/chipidea/otg.h
index 1c9ad1a..b8b4bbd5 100644
--- a/drivers/usb/chipidea/otg.h
+++ b/drivers/usb/chipidea/otg.h
@@ -17,6 +17,7 @@ int ci_hdrc_otg_init(struct ci_hdrc *ci);
 void ci_hdrc_otg_destroy(struct ci_hdrc *ci);
 enum ci_role ci_otg_role(struct ci_hdrc *ci);
 void ci_handle_vbus_change(struct ci_hdrc *ci);
+void ci_handle_id_switch(struct ci_hdrc *ci);
 void ci_handle_vbus_connected(struct ci_hdrc *ci);
 static inline void ci_otg_queue_work(struct ci_hdrc *ci)
 {
-- 
1.7.5.4

